### ISSUE TITLE
Asking for a LAZYMODULEMIXIN warning

### ISSUE THREAD

**Issue by manuel-dileo** [pytorch/pytorch#94132]
Hi, I spent several days trying to understand why a gcn model (from pytorch geometric) produced different results on different runs, despite the manual seed setting.

The answer was due to the way lazy modules in PyTorch initialize parameters differently than other modules in a same ML model. (as described here: https://pytorch.org/docs/stable/generated/torch.nn.modules.lazy.LazyModuleMixin.html ). To mitigate this unpleasant situation, could you add a warning about reproducibility of experiments when using a lazy module? Thank you in advance

**Comment by ezyang** [comment]
The documentation seems to cover the gotcha. What kind of warning would you be looking for?
